velox/connectors/hive/HiveDataSource.cpp

```diff
@@ -702,6 +704,8 @@ void HiveDataSource::setFromDataSource(
 
   split_ = std::move(source->split_);
   if (source->splitReader_ && source->splitReader_->emptySplit()) {
+    runtimeStats_.skippedSplits += source->runtimeStats_.skippedSplits;
```

Comment:
This will be double counting, no?  It already added in `SplitReader::prepareSplit`

Reply:
These metrics are collected within the `prepareSplit` function.
The purpose of this section of code is to pass the metrics obtained during the 'preloading' data source (captured by `runtimeStats_`) to the main data source (`dataSource_`).
This is found by new added UT. Without this line, the result of `prepareSplit` in new UT is consistently 0, whereas the expected result is 4. 
The correction ensures that the runtime metrics are properly transferred to `dataSource_`, leading to the desired outcome.

Thanks.